### PR TITLE
Update golang to 1.17

### PR DESCRIPTION
**Pull request by xinyanw409**
**What this PR does / why we need it**:
Golang 1.16 has a critical vulnerability, therefore upgrading the version to 1.17
**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Upgrade golang version to 1.17
```

### CODE REVIEW

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/683/

**Comment by svcbot-qecnsdp** [comment]
Sorry, some of the pre-check tests are failing for this PR ! 

Please check the Jenkins job for more details
https://container-dp.svc.eng.vmware.com/job/CNSDP-CI/684/

File: go.mod [hunk 1]
```diff
@@ -25,6 +23,108 @@ require (
 	k8s.io/utils v0.0.0-20210707171843-4b05e18ac7d9
 )
 
+require (
+	cloud.google.com/go v0.54.0 // indirect
```
Comment (lintongj) [inline go.mod:27]: Why do we need to add so many dependencies? Can you run `go mod tidy` before pushing to remote?

**Review by lintongj — COMMENTED** [review]
Please update the PR title accordingly.

File: go.mod (hunk 1 above)
Comment (xinyanw409) [reply go.mod:27]: Those changes are added automatically after running ```go mod tidy``` @lintongj

**Review by lintongj — APPROVED** [review]

**Review by xing-yang — APPROVED** [review]
LGTM

**Pull request merged**